### ISSUE TITLE
Duplicate records when traversing knowledge base via API

### ISSUE THREAD

**Issue by lukefan** [langgenius/dify#10252]
### Self Checks

- [X] This is only for bug report, if you would like to ask a question, please head to [Discussions](https://github.com/langgenius/dify/discussions/categories/general).
- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Dify version

0.10.2

### Cloud or Self Hosted

Self Hosted (Docker)

### Steps to reproduce

curl --location --request GET 'http://localhost/v1/datasets/{dataset_id}/documents?page=${page}&limit=100' \
--header 'Authorization: Bearer {api_key}'

### ✔️ Expected Behavior

All non - repeated document information.

### ❌ Actual Behavior

I am traversing all the documents in the knowledge base through an API. There are 1000 documents in the knowledge base and I fetch them 100 at a time, page by page. However, I have obtained a large number of duplicate records. The IDs and file names are all duplicated. This issue is causing inefficiency and confusion in my work. I would appreciate it if this problem could be addressed.

**Comment by lukefan** [comment]
I am continuously calling the API and fetching the next page's content as long as has_more is true. The occurrence of duplicate records is definitely an issue with the API design, especially since the IDs are also duplicated.
Currently, I am not sure if I have fetched all the articles. It seems that some article titles are available on the dify website but cannot be found in the fetched library.
The total number of fetched records with duplicates is consistent with the total number of articles in the knowledge base. Therefore, I suspect that some articles have not been fetched.
I feel that it might be due to inconsistent sorting methods in the database each time the fetching occurs. As a result, different pages may contain duplicate content.

**Comment by tjlindeman** [comment]
I encountered this issue as well. My suspicion is that the KB API is unable to handle any requests that exceed 100 records, regardless of whether you correctly page the request. For example, I have a KB with 101 documents. I set [${page}&limit=20. For pages 1-5 it returns unique records. For page 6 it returns a random duplicate record present in a preceding page.

I guess I will have to manage my KBs so that none of my KBs has greater than 100 documents. And none of my documents has more than 100 chunks.

Any advice?
